metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery.py

```diff
@@ -1189,7 +1202,19 @@ def get_tables_for_dataset(
                     self.report.report_dropped(table_identifier.raw_table_name())
                     continue
 
-                if table_count % self.config.number_of_datasets_process_in_batch == 0:
+                table_count_in_this_batch += 1
+
+                if (
+                    table_count_in_this_batch
+                    % self.config.number_of_datasets_process_in_batch
+                    == 0
+                ) or (
+                    partitioned_table_count_in_this_batch > 0
+                    and partitioned_table_count_in_this_batch
+                    % self.config.number_of_partitioned_datasets_process_in_batch
+                    == 0
+                ):
```

Comment:
Could we replace the counting with `table_count_in_this_batch` by just checking `len(table_items)`? It seems like they'll match each other, unless we're getting duplicate `table.table_id` keys, and in any case, that seems like the number we want to track when making the query

Also, since we're resetting `partitioned_table_count_in_this_batch = 0` each query, I think the second conditional can be simplified to `partitioned_table_count_in_this_batch == self.config.number_of_partitioned_datasets_process_in_batch`